courseworks/week1/XU_Shiqinrui/Assignment 1.md

```diff
@@ -0,0 +1,57 @@
+### Q1
+The result of `lscpu` is shown below:
+![[Pasted image 20230214222734.png]]
+The estimation is:
+$$
+\begin{aligned}
+FLPOS &= number\ of\ cores \times frequency\times4\times2\\
+&=16\times3.7GHz\times4\times2\\
+&=473.6GFLPOS
+\end{aligned}
+$$
+### Q2
+The signal can't travels faster than light. And the signal in a real circuit is actually much slower than the speed of light in vaccum. The fomula to calculate the actual speed is $v_s=\frac{c}{\sqrt{\epsilon_r\mu_r}}$. Since I didn't find a convincing result of speed of light in copper, I'll assume that the speed of light in copper is $2\times 10^8m/s$. And if CPU wants some data from memory, it needs to send the request to memory first. Then the memory will send the data to CPU. Thus the whole length the signal nees to travel is $10cm\times2=2\times10^{-1}m$. So the shortest time should be:
+$$\begin{aligned}
+t&=2\times10^{-1}m\div2\times 10^8m/s\\
+&=1ns
+\end{aligned}$$
+But except the travel time, it also takes time that memory to look up the corresponeding data. And there are also some other factors that can slow down the process. Thus the actual  latency could be tens of $ns$ to hundred $ns$.
```

Comment:
you may also need to calculate the clock cycle of you CPU and compare the difference.